src/libsync/propagateuploadng.cpp

```diff
@@ -293,15 +293,14 @@ void PropagateUploadFileNG::slotPropfindFinished()
     slotJobDestroyed(job); // remove it from the _jobs list
     propagator()->_activeJobList.removeOne(this);
 
-    _currentChunk = _rangesToUpload.first().start;
+    _currentChunk = 0;
     _sent = 0;
-    while (_serverChunks.contains(_currentChunk)) {
-        _sent += _serverChunks[_currentChunk].size;
-        if (updateRanges(_currentChunk, _serverChunks[_currentChunk].size))
-            _serverChunks.remove(_currentChunk);
+
+    for (auto& chunkId : _serverChunks.keys()) {
+        if (updateRanges(chunkId, _serverChunks[chunkId].size))
```

Comment:
Yes, that looks right. The _rangesToUpload get adjusted for the server data and any unneeded server data gets deleted further down in this function. Looks good.

This will need thorough unittests later.